### PR TITLE
[ACR] Address docstring

### PR DESCRIPTION
**Pull request by YalinLi0312**
For issue https://github.com/Azure/azure-sdk-for-python/issues/22250

Address docstring for optional variables.

### CODE REVIEW

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py [hunk 1]
```diff
@@ -31,17 +31,17 @@ class ArtifactManifestProperties(object):  # pylint: disable=too-many-instance-a
...
-    :vartype last_updated_on: ~datetime.datetime
+    :vartype last_updated_on: Optional[~datetime.datetime]
     :ivar operating_system: Operating system for the artifact.
-    :vartype operating_system: ~azure.containerregistry.ArtifactOperatingSystem
-    :ivar str repository_name: Repository name the artifact belongs to.
-    :ivar str size_in_bytes: Size of the artifact.
-    :ivar List[str] tags: Tags associated with a registry artifact.
+    :vartype operating_system: Optional[~azure.containerregistry.ArtifactOperatingSystem]
+    :ivar Optional[str] repository_name: Repository name the artifact belongs to.
+    :ivar Optional[int] size_in_bytes: Size of the artifact.
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py:43]: Are you sure that this one is just a docs fix, and that the model itself is returning an `int`?

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py [hunk 2]
```diff
@@ -31,17 +31,17 @@ class ArtifactManifestProperties(object):  # pylint: disable=too-many-instance-a
     :ivar bool can_read: Read Permissions for an artifact.
     :ivar bool can_list: List Permissions for an artifact.
     :ivar architecture: CPU Architecture of an artifact.
-    :vartype architecture: ~azure.containerregistry.ArtifactArchitecture
+    :vartype architecture: Optional[~azure.containerregistry.ArtifactArchitecture]
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py:34]: What did you reference to determine that all these properties were `Optional`?

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py (hunk 1 above)
Comment (YalinLi0312) [reply sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py:43]: 1. The property type should be consistent in model return type and docstring.
2. AFAIK the doc portal is referenced by the docstring here. @kristapratico Can u help to confirm?

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py (hunk 2 above)
Comment (YalinLi0312) [reply sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py:34]: From line 48.

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py (hunk 1 above)
Comment (kristapratico) [reply sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py:43]: Yes, the docs.microsoft.com ref docs will use these docstrings

**Review by annatisch — APPROVED** [review]

**Pull request merged**